inference-engine/src/gna_plugin/gna_model_serial.cpp

```diff
@@ -77,12 +77,13 @@ const int gna_header_magic = is_little_endian() ?  0x4d414e47 : 0x474e414d;
 
 GNAPluginNS::HeaderLatest::ModelHeader GNAModelSerial::ReadHeader(std::istream &is) {
     is.exceptions(std::istream::failbit);
+    auto startPos = is.tellg();
     is.seekg(0, is.end);
-    auto stream_len = is.tellg();
+    auto stream_len = is.tellg() - startPos;
     if (stream_len == -1) {
```

Comment:
I suppose this check should be moved higher

Reply:
I meant that there is a sense to check startPos at first

Reply:
Fixed. Unit test was also implemented for this